client/db/src/lib.rs

```diff
@@ -515,12 +515,18 @@ impl<Block: BlockT> sc_client_api::blockchain::Backend<Block> for BlockchainDb<B
 						)),
 					},
 					TransactionStorageMode::StorageChain => {
-						match Vec::<Block::Hash>::decode(&mut &body[..]) {
-							Ok(hashes) => {
-								let extrinsics: ClientResult<Vec<Block::Extrinsic>> = hashes.into_iter().map(
-									|h| self.extrinsic(&h).and_then(|maybe_ex| maybe_ex.ok_or_else(
-										|| sp_blockchain::Error::Backend(
-											format!("Missing transaction: {}", h))))
+						match Vec::<(Block::Hash, Vec<u8>)>::decode(&mut &body[..]) {
+							Ok(index) => {
+								let extrinsics: ClientResult<Vec<Block::Extrinsic>> = index.into_iter().map(
+									|(hash, mut data)| {
+										if let Some(mut t) = self.indexed_transaction(&hash)? {
+											data.append(&mut t);
+										}
+										Block::Extrinsic::decode(&mut data.as_slice())
```

Comment:
If we would use a custom `Input` here, we could prevent calling `append` above and allocating a third time.
This custom input would just combine `data` and `t`